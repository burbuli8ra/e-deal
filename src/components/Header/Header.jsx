import React from 'react';
import Styled from './Header.style';

const Header = () => {
  return (
    <Styled.Header data-testid="header">
      <Styled.Logo data-testid="header-logo" to="/">
        <Styled.Icon data-testid="header-icon" />
        eDeal
      </Styled.Logo>
    </Styled.Header>
  );
};

export default Header;