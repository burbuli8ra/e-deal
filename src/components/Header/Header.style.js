import styled from '@emotion/styled';
import { Link } from 'react-router-dom';
import Assets from 'assets';
import { colors } from 'theme';

const Header = styled.header`
  background-color: ${colors.white};
  box-shadow: 0 0 9px 0 rgba(140, 140, 158, 0.25);
  padding: 20px;
  text-align: center;
`;
Header.displayName = 'Header';

const Icon = styled(Assets.Logo)`
  height: 28px;
  width: 28px;
  margin-right: 4px;
`;
Icon.displayName = 'Icon';

const Logo = styled(Link)`
  align-items: center;
  color: ${colors.pastelBlue};
  display: inline-flex;
  font-size: 24px;
  font-weight: bold;
  text-decoration: none;
`;
Logo.displayName = 'Logo';

export default {
  Header,
  Icon,
  Logo
}