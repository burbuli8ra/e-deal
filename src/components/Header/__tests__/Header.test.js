import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import Header from '../Header';

const setup = props => (
  <Router>
    <Header {...props} />
  </Router>
);

describe('Header component', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {};
    wrapper = render(setup(props));
  });

  test('should match snapshot', () => {
    const { asFragment } = wrapper;
    expect(asFragment()).toMatchSnapshot();
  });
});