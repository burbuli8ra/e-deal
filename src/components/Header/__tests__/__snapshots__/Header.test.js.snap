// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot 1`] = `
<DocumentFragment>
  <header
    class="css-3vq7ia"
    data-testid="header"
  >
    <a
      class="css-1rzlx2v"
      data-testid="header-logo"
      href="/"
    >
      <svg
        class="css-xzn57i"
        data-testid="header-icon"
      >
        logo.svg
      </svg>
      eDeal
    </a>
  </header>
</DocumentFragment>
`;
